packages/solidity-contracts/deploy/beta5/005.fuel_message_portal_v2.ts

```diff
@@ -0,0 +1,40 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { DeployFunction } from 'hardhat-deploy/dist/types';
+
+import { FuelMessagePortalV2__factory as FuelMessagePortalV2 } from '../../typechain';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {
+    ethers,
+    upgrades: { upgradeProxy, erc1967 },
+    deployments: { get, save },
+  } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const fuelMessagePortal = await get('FuelMessagePortal');
+
+  const { deployTransaction, address } = await upgradeProxy(
+    fuelMessagePortal,
+    new FuelMessagePortalV2(deployer),
+    {
+      unsafeAllow: ['constructor'],
+      constructorArgs: [ethers.constants.MaxUint256],
+    }
+  );
+
+  await deployTransaction.wait();
+  const implementation = await erc1967.getImplementationAddress(address);
+
+  console.log('Upgraded FuelMessagePortal at', address);
+  await save('FuelMessagePortal', {
+    address,
+    abi: [],
+    implementation,
+  });
+
+  return true;
+};
+
+func.tags = ['upgrade_portal_v2', 'message_portal_v2', 'FuelMessagePortalV2'];
+func.id = 'fuel_message_portal_v2_upgrade';
```

Comment:
What's the purpose of the tags and id?

Reply:
Tags: so that you can do something like `npx hardhat deploy --network beta5 --tags [comma separated list of tags]`. `hardhat-deploy` is a well known plugin in the space, a very popular one for handling the deployments. So it replaces our very handmade deploy scripts.

ID: when you run a script, the ID is the marker that prevents us from running the same script twice / overriding an upgrade, which could be a major mistake that is not currently well covered by our handmade deploy scripts.